### DISCUSSION TITLE
Is the "base" configuration must be the same between router and vite?What if I need different configuration?

### DISCUSSION THREAD

**Discussion by dang0705** [vuejs/router#1613] [Help and Questions] [answered]
### What problem does this feature solve?
In router3, I can set different base between router and vite for multiple app with same static sources like js, css eg:

in router 
base: '/event/1/'  or '/event/2/' '/event/3/'  ....

in vite.config.js   
base :'/event/'

Why to do this? Because I need the event-id like '1','2','3' from the location.pathname, and it'ill be a part of router base like '/event/1/', and the packaged code's static sources path are all in the path '/event/', so it very easy to config differently in router3
But in router4, they must be the same , or the static resources couldn't be resolve in development mode

### What does the proposed API look like?
The router4's base config like router3, can be different with vite's base, and the resources can be reslove correctly

**Comment by posva** [comment, marked as answer]
That is not possible, `base` is to define a serving folder : https://router.vuejs.org/api/interfaces/RouterHistory.html#Properties-base

You can define a _prefix_ in routes with

```ts
routes: [
	{ path: '/prefix', children: [{ path: '', name: 'Home' }, { path: 'about', name: 'About' }] }
]
```
